src/plugins/intel_gna/tests/unit/pre_post_processing/gna_input_preproc_test.cpp

```diff
@@ -0,0 +1,279 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <cmath>
+#include <memory>
+#include <vector>
+
+#include "common_test_utils/data_utils.hpp"
+#include "common_test_utils/ngraph_test_utils.hpp"
+#include "pre_post_process/converter_factory.hpp"
+#include "pre_post_process/input_output_data_handler.hpp"
+
+using namespace InferenceEngine;
+using namespace pre_post_processing;
+
+namespace testing {
+
+typedef std::tuple<Precision,                // input precision
+                   SizeVector,               // input shape
+                   intel_dnn_orientation_t,  // orientation
+                   float,                    // scale factor
+                   bool,                     // gna device
+                   bool,                     // set low precision
+                   bool,                     // test avx2
+                   uint32_t                  // input range
+                   >
+    GNAInputPrecisionParams;
+
+template <typename U, typename T>
+class GNAInputPrecisionTest : public ::testing::TestWithParam<GNAInputPrecisionParams> {
+public:
+    void SetUp() override {
+        uint32_t input_range;
+        std::tie(prc, shape, orientation, sf, is_gna_device, is_low_precision, test_avx2, input_range) = GetParam();
+        input_vals.resize(ov::shape_size(shape));
+        CommonTestUtils::fill_data_random(&input_vals[0], ov::shape_size(shape), input_range);
+
+        std::transform(begin(input_vals), end(input_vals), std::back_inserter(refer_vals), [this](U i) {
+            return round(i * sf);
+        });
+
+        if (test_avx2) {
+            ConverterFactory converter_factory;
+            auto converter = converter_factory.create_converter();
+            if (converter == nullptr) {
```

Comment:
what about a test for factory to ensure that it returns converter in case avx2 is enabled and available and to check if converter is not returned in case avx2 was enabled but not available in runtime.